File: filmforum_backend/src/main/java/rs/ac/bg/fon/filmforum_backend/dto/MovieDTO.java
package rs.ac.bg.fon.filmforum_backend.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PositiveOrZero;
import rs.ac.bg.fon.filmforum_backend.domain.Genre;

import java.util.List;

public record MovieDTO(
        long id,
        boolean adult,
        String backdropPath,
        @PositiveOrZero(message = "Movie budget cannot be a negative number")
        long budget,
        List<Genre> genres,
        String homepage,
        String imdb,
        String originalLanguage,
        String originalTitle,
        @NotBlank(message = "Movie overview cannot be blank")
        String overview,
        double popularity,
        String posterPath,
        String releaseDate,
        long revenue,
        @PositiveOrZero(message = "Movie runtime cannot be a negative number")
        int runtime,
        String status,
        @NotBlank(message = "Movie tagline cannot be blank")
        String tagline,
        String title,
        boolean video,
        List<ActorDTO> cast,
        List<CrewMemberDTO> crew
) {
}

File: filmforum_mobile/src/domain/MovieDTO.ts
import {ActorDTO} from './ActorDTO';
import {CrewMemberDTO} from './CrewMemberDTO';
import {Genre} from "./Genre";

export interface MovieDTO {
    id: number;
    adult: boolean;
    backdropPath: string;
    budget: number;
    genres: Genre[];
    homepage: string;
    imdb: string;
    originalLanguage: string;
    originalTitle: string;
    overview: string;
    popularity: number;
    posterPath: string;
    releaseDate: string;
    revenue: number;
    runtime: number;
    status: string;
    tagline: string;
    title: string;
    video: boolean;
    cast: ActorDTO[];
    crew: CrewMemberDTO[];
}
